prusti-encoder/src/encoders/ty/generics/traits.rs

```diff
@@ -0,0 +1,91 @@
+use prusti_rustc_interface::{middle::ty::AssocKind, span::def_id::DefId};
+use task_encoder::{EncodeFullResult, TaskEncoder, TaskEncoderDependencies};
+use vir::{FunctionIdn, vir_format_identifier};
+
+use crate::encoders::ty::generics::{GParams, GenericParamsEnc};
+
+pub struct TraitEnc;
+
+#[derive(Debug, Clone, Copy)]
+pub struct TraitData<'vir> {
+    pub trait_name: &'vir str,
+    pub type_did_fun_mapping: &'vir [(DefId, FunctionIdn<'vir, vir::ManyTyVal, vir::TyVal>)],
```

Comment:
The "public" interface (i.e., when `GenericParams` are using this) should rather be an actual map, or perhaps a getter function `(&self, DefId) -> FunctionIdn<...>`. As an added bonus, when constructing such a map, you can easily make sure you do not have multiple entries for the same `DefId` (the sanity check that you are performing in `GenericParams` right now).